### PR TITLE
Render image with camera_pos, target_pos, camera_up vector as arguments

### PR DESCRIPTION
**Pull request by hellojas**
- adding internal set vector as helper to renderImage with three vectors
- ability to call renderImage with three vectors: camera position, target position, and up vector

successfully tested by comparing view and projection matrix generated from C-API and python camera.py code:

➜  bullet_experiments python test_renderImage.py

camera pos:
 2.000000 0.000000 1.000000
targetPos pos:
 0.000000 0.000000 0.000000
cameraUp pos:
 0.000000 0.000000 1.000000

printing python viewMatrix:
[[ 0.         -0.4472136   0.89442719  0.        ]
 [ 1.          0.         -0.          0.        ]
 [-0.          0.89442719  0.4472136   0.        ]
 [-0.         -0.         -2.23606798  1.        ]]
printing python projectionMatrix
[[ 1.         0.         0.         0.       ]
 [ 0.         1.         0.         0.       ]
 [ 0.         0.        -1.00002   -1.       ]
 [ 0.         0.        -0.0200002  0.       ]]

printing b3RequestCameraImageSetViewMatrix:

 0.000000 -0.447214 0.894427 0.000000
 1.000000 0.000000 -0.000000 0.000000
 -0.000000 0.894427 0.447214 0.000000
 -0.000000 -0.000000 -2.236068 1.000000

printing b3RequestCameraImageSetProjectionMatrix:

 1.000000 0.000000 0.000000 0.000000
 0.000000 1.000000 0.000000 0.000000
 0.000000 0.000000 -1.000002 -1.000000
 0.000000 0.000000 -0.002000 0.000000

### CODE REVIEW

**Comment by erwincoumans** [comment]
Thanks for the pull request.
It would be great if you also expose the near/far instead of only relying on hard-coded values. It is easy to show in examples (and docs) some meaningful default values.

Some checks failed: just move the 'int i' out of the 3 for loops, this is C99: (clock on the 'Details' next to the red cross:
/home/travis/build/bulletphysics/bullet3/examples/pybullet/pybullet.c:1084:17: error: ‘for’ loop initial declarations are only allowed in C99 mode.

**Comment by hellojas** [comment]
near/far values exposed by calling 

    img_arr = pybullet.renderImage(
        w, h, camera_pos, target_pos, up, near, far)

commented out debugger printf statements which caused c99 error. 

thanks!

**Comment by erwincoumans** [comment]
Thanks for the contribution, this API makes it much easier to render an image, without having to worry how to assemble the view and projection matrices.

File: examples/pybullet/pybullet.c [hunk 1]
```diff
@@ -1001,6 +1038,47 @@ static PyObject* pybullet_renderImage(PyObject* self, PyObject* args)
        }
    }
 
+   if (size==5) // set camera resoluation and view and projection matrix
```
Comment (matpalm) [inline examples/pybullet/pybullet.c]: size==7

File: examples/pybullet/pybullet.c [hunk 2]
```diff
@@ -1396,6 +1474,13 @@ static PyMethodDef SpamMethods[] = {
     ////todo(erwincoumans)
     //collision info
     //raycast info
+        
+    //applyBaseForce
+    //applyLinkForce
+     
+	{"renderImage", pybullet_renderImage, METH_VARARGS,
+	"Render an image (given the pixel resolution width, height, view matrix, projection matrix, near and far vlues), and return the 8-8-8bit RGB pixel data and floating point depth values"},
```
Comment (matpalm) [inline examples/pybullet/pybullet.c]: s/vlues/values/

File: examples/pybullet/pybullet.c [hunk 3]
```diff
@@ -1396,6 +1474,13 @@ static PyMethodDef SpamMethods[] = {
     ////todo(erwincoumans)
     //collision info
     //raycast info
+        
+    //applyBaseForce
+    //applyLinkForce
+     
+	{"renderImage", pybullet_renderImage, METH_VARARGS,
```
Comment (matpalm) [inline examples/pybullet/pybullet.c]: this is being defined twice now (see up 20 lines)

**Comment by hellojas** [comment]
Ah, thanks Mat! Will fix in subsequent pull req!

**Pull request merged**